core/src/main/scala/kafka/log/Log.scala

```diff
@@ -347,14 +367,16 @@ class Log(val dir: File,
           } catch {
             case e: IOException => throw new KafkaException("Error in validating messages while appending to log '%s'".format(name), e)
           }
-          validMessages = validatedMessages
+          validMessages = validateAndOffsetAssignResult.validatedMessages
+          maxTimestampInMessageSet = validateAndOffsetAssignResult.maxTimestamp
```

Comment:
Could we just set appendInfo.maxTimestamp here and get rid of maxTimestampInMessageSet?